Versioneer is failing on TravisCI

Versioneer needs `git describe` to be able to find an annotated tag in the relevant branch and this is failing, [here](https://travis-ci.com/github/numpy/numpy/jobs/481119639). I suspect that the problem is caused by the limited clone depth used by TravisCI (50) but don't quite know how to interpret that as the most recent tag is already 689 commits back and the failures started yesterday.

Copying the failure for future searchability:
```
___________________________ test_valid_numpy_version ___________________________
6260
6261    def test_valid_numpy_version():
6262        # Verify that the numpy version is a valid one (no .post suffix or other
6263        # nonsense).  See gh-6431 for an issue caused by an invalid version.
6264        version_pattern = r"^[0-9]+\.[0-9]+\.[0-9]+(|a[0-9]|b[0-9]|rc[0-9])"
6265        dev_suffix = r"\.dev0\+[0-9]*\.g[0-9a-f]+"
6266        if np.version.release:
6267            res = re.match(version_pattern, np.__version__)
6268        else:
6269            res = re.match(version_pattern + dev_suffix, np.__version__)
6270    
6271>       assert_(res is not None, np.__version__)
6272E       AssertionError: 0+untagged.1185.g7e82a56
```

--- a/.travis.yml
+++ b/.travis.yml
@@ -16,6 +16,10 @@ addons:
       # Speedup builds, particularly when USE_CHROOT=1
       - eatmydata
 
+# Disable clone depth
+git:
+  depth: false
+
 cache:
   directories:
     - $HOME/.cache/pip

